Source/buildimplementationcpp.go

```diff
@@ -691,6 +693,80 @@ func buildCPPInterfaceWrapperMethods(component ComponentDefinition, class Compon
 	return nil
 }
 
+func writeCImplementsInterfaceMethod(component ComponentDefinition, method ComponentDefinitionMethod, w LanguageWriter, NameSpace string, IBaseClassName string) error {
+	cParams, err := GenerateCParameters(method, "", NameSpace)
+	if err != nil {
+		return err
+	}
+
+	cparameters := ""
+	for _, cParam := range cParams {
+		if cparameters != "" {
+			cparameters = cparameters + ", "
+		}
+		cparameters = cparameters + cParam.ParamType + " " + cParam.ParamName
+	}
+
+	CMethodName := fmt.Sprintf("%s_%s", strings.ToLower(NameSpace), strings.ToLower(method.MethodName))
+
+	classHashMap := make(map[string][]ComponentDefinitionClass)
+
+	for i := 0; i < len(component.Classes); i++ {
+		class := component.Classes[i]
+		key := fmt.Sprintf("%02X", class.classHash()[0])
+		classHashMap[key] = append(classHashMap[key], class)
+	}
+
+	w.Writeln("")
+	w.Writeln("/*************************************************************************************************************************")
+	w.Writeln("  %s", method.MethodDescription)
+	w.Writeln("**************************************************************************************************************************/")
+	w.Writeln("")
+
+	w.Writeln("%sResult %s(%s)", NameSpace, CMethodName, cparameters)
+	w.Writeln("{")
+	w.Writeln("  if (nClassHashBufferSize != %d) // Hash length must be as expected", md5.Size)
+	w.Writeln("    return RTTI_ERROR_INVALIDPARAM;")
+	w.Writeln("")
+	w.Writeln("  %s* pIBaseClassInstance = (%s *)pObject;", IBaseClassName, IBaseClassName)
+	w.Writeln("")
+	w.Writeln("  switch(pClassHashBuffer[0]) {")
+
+	keys := make([]string, 0, len(classHashMap))
+	for key := range classHashMap {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
+
+	for i := range keys {
+		w.Writeln("    case 0x%s:", keys[i])
+		classes := classHashMap[keys[i]]
+		for j := range classes {
+			class := classes[j]
+			hash := class.classHash()
+			w.BeginLine()
+			w.Printf("      static const RTTI_uint8 s_%sHash[] = {", class.ClassName)
```

Comment:
hard coded namespace "RTTI" should be replaced with "componentdefinition.namespace"

Same in line 757 and potentially more.